### PR TITLE
Master wowl dates boi

### PR DESCRIPTION
**Pull request by brboi**
date utils: create sections ?? + date utils confirm => isValidDateTime + there's a comment about replacing returns false with DateTime.invalid(reason)

### CODE REVIEW

File: addons/web/static/src/core/l10n/dates.js [hunk 1]
```diff
@@ -40,40 +47,102 @@ const normalizeFormatTable = {
...
+
+/**
+ * Enforces some restrictions to a Luxon DateTime object.
+ * Returns it if within those restrictions.
+ * Returns false otherwise.
+ *
+ * @param {DateTime | false} date
+ * @returns {DateTime | false}
+ */
+function restrain(dt) {
```
Comment (ged-odoo) [inline addons/web/static/src/core/l10n/dates.js]: i think `constrain` would be more appropriate

**Comment by ged-odoo** [comment]
robodoo r+

**Comment by ged-odoo** [comment]
@brboi runbot is red

**Comment by brboi** [comment]
> @brboi runbot is red

Yes...

I had an issue with https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test#using_test_on_a_regex_with_the_global_flag

Solved.

@ged-odoo I also changed the DateTime.prototype.toJSON. See last item in commit message

No other changes since your last review than lines:
- 14
- 131

**Comment by ged-odoo** [comment]
robodoo r+

**Pull request closed without merging**